game-core/src/main/java/games/strategy/engine/data/unit/ability/CombatUnitAbility.java

```diff
@@ -0,0 +1,114 @@
+package games.strategy.engine.data.unit.ability;
+
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.delegate.battle.BattleState;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NonNull;
+import lombok.experimental.FieldDefaults;
+import lombok.experimental.NonFinal;
+
+/** Unit Ability for combat situations */
+@Data
+@FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+@Builder(toBuilder = true)
+public class CombatUnitAbility {
+
+  public static final CombatUnitAbility EMPTY = CombatUnitAbility.builder().name("Empty").build();
+
+  enum DiceType {
+    // Use AA dice (attackAA, offensiveAttackAA)
+    AA,
+    // Use bombard dice (bombard or attack if bombard is empty)
+    BOMBARD,
+    // Use normal dice (attack, defense)
+    NORMAL
+  }
+
+  /** The name of this unit ability that will be shown in the Battle UI */
+  @NonNull String name;
+
+  /** The unit types that have this ability */
+  @Builder.Default @NonFinal Collection<UnitType> attachedUnitTypes = List.of();
```

Comment:
This class will also be serialized as well during battles.  So it isn't because of serialization.

The reason is that each player can have different unit abilities for the same unit type.  But unit types are player agnostic.  So, I either need to make the unit abilities store the player their attached to or store the unit type that they are attached to.  I went with the later because I could put the player <-> unit ability mapping in the battle phase and not have to touch the player class.

I didn't call it `unitTypes` because `targets` is also a collection of `UnitType`s and thought that having one member called `unitTypes` and another called `targets` would be confusing since both are just collection of `UnitType`s.  I wanted to be more explicit in the name to ease identification.